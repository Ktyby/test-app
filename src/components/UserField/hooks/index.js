import useHandleDeleteUser from "./useHandleDeleteUser";
import useHandleEditUser from "./useHandleEditUser";
import useModal from "./useModal";

export { useHandleDeleteUser, useModal, useHandleEditUser };
