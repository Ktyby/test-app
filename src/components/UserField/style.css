.user-field {
  display: flex;
  justify-content: space-between;
  align-items: center;

  width: 800px;
}

.user-cell {
  color: #fff;
  width: 200px;
}

.user-cell__button {
  width: 150px;
  height: 35px;

  border: none;
  border-radius: 10px;

  cursor: pointer;
  transition: 0.4s;
}

.user-field__button:hover {
  background-color: #000;
  color: #fff;
}
