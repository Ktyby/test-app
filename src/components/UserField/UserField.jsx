import React, { useState } from "react";
import PropTypes from "prop-types";

import "./style.css";

import ConfirmationModal from "../ConfirmationModal";
import Modal from "../Modal";
import Form from "../Form";

import { useHandleDeleteUser, useModal, useHandleEditUser } from "./hooks";

const UserField = ({ user }) => {
  const [displayModal, setDisplayModal] = useState(false);
  const [displayConfirmation, setDisplayConfirmation] = useState(false);

  const { handleModalShow, handleModalClose } = useModal({
    setDisplayModal,
  });

  const handleDeleteUser = useHandleDeleteUser({
    setDisplayConfirmation,
  });

  const handleEditUser = useHandleEditUser({
    handleModalClose,
  });

  return (
    <tr>
      <td className="user-cell">{user.surname}</td>
      <td className="user-cell">{user.name}</td>
      <td className="user-cell">{user.birthday}</td>
      <td className="user-cell">{user.email}</td>
      <td className="user-cell">
        <button
          className="user-cell__button"
          value={user.id}
          onClick={handleModalShow}
        >
          Edit
        </button>
      </td>
      <td className="user-cell">
        <button
          className="user-cell__button"
          onClick={() => setDisplayConfirmation(true)}
        >
          Delete
        </button>
      </td>
      {displayConfirmation && (
        <ConfirmationModal
          onDeleteUser={handleDeleteUser}
          onClose={() => {
            setDisplayConfirmation(false);
          }}
        />
      )}
      {displayModal && (
        <Modal onClose={handleModalClose} title="Edit user">
          <Form userData={user} onSubmit={handleEditUser} />
        </Modal>
      )}
    </tr>
  );
};

UserField.propTypes = {
  user: PropTypes.object,
};
UserField.defaultProps = {
  user: {},
};

export default UserField;
