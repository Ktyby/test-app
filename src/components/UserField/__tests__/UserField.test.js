import renderer from "react-test-renderer";
import React from "react";
import { Provider } from "react-redux";
import configureMockStore from "redux-mock-store";

import UserField from "../UserField";

const mockStore = configureMockStore();

describe("components => UserField", () => {
  const store = mockStore({
    users: [],
  });

  const component = renderer.create(
    <Provider store={store}>
      <UserField
        user={{
          name: "Max",
          surname: "Verstappen",
          birthday: "2022-08-07",
          email: "max.verstappen@gmail.com",
          id: "some id string",
        }}
      />
    </Provider>
  );

  it("should render UserField component", () => {
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
