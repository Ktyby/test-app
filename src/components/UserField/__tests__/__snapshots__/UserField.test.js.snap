// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components => UserField should render UserField component 1`] = `
<tr>
  <td
    className="user-cell"
  >
    Verstappen
  </td>
  <td
    className="user-cell"
  >
    Max
  </td>
  <td
    className="user-cell"
  >
    2022-08-07
  </td>
  <td
    className="user-cell"
  >
    max.verstappen@gmail.com
  </td>
  <td
    className="user-cell"
  >
    <button
      className="user-cell__button"
      onClick={[Function]}
    >
      Edit
    </button>
  </td>
  <td
    className="user-cell"
  >
    <button
      className="user-cell__button"
      onClick={[Function]}
    >
      Delete
    </button>
  </td>
</tr>
`;
